keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/diff/LogProbGradients.java

```diff
@@ -0,0 +1,51 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.diff;
+
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.VertexId;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class LogProbGradients {
+
+    private final Map<VertexId, DoubleTensor> partials;
+
+    public LogProbGradients() {
+        this.partials = new HashMap<>();
+    }
+
+    public LogProbGradients add(LogProbGradients addition) {
+        return add(addition.partials);
+    }
+
+    public LogProbGradients add(Map<VertexId, DoubleTensor> addition) {
+
+        Map<VertexId, DoubleTensor> initial = partials;
+
+        for (Map.Entry<VertexId, DoubleTensor> entry : addition.entrySet()) {
+            VertexId id = entry.getKey();
+            if (initial.containsKey(id)) {
+
```

Comment:
nit: this looks a bit funky just delete the white space

Reply:
I've refactored this a bit. Kept some whitespace but also improved it a bit.